lib/utils.pm

```diff
@@ -258,6 +259,21 @@ sub get_netboot_mirror {
     return get_var('MIRROR_' . uc($m_protocol));
 }
 
+sub zypper_call {
+    my $command = shift;
+    my $timeout = shift || 700;
+    my $str     = bmwqemu::hashed_string("ZN$command");
+
+    script_run("zypper -n $command; echo $str-\$?- > /dev/$serialdev", 0);
+
+    my $ret = wait_serial(qr/$str-\d+-/, $timeout);
+
+    if ($ret =~ /$str-(\d+)/) {
+        return $1;
+    }
```

Comment:
No it doesn't. It returns only \d+ part. That's due to return itself is able to cope with array returns.

Anyway, I suggest to use the second example in my previous comment.